### PR TITLE
feat(material-experimental): add test harness for tabs

### PR DESCRIPTION
**Pull request by devversion**
Resolves COMP-193.

### CODE REVIEW

File: src/material-experimental/mdc-tabs/harness/tabs-harness.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  static hostSelector = '.mat-tab-group';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a radio-button with
+   * specific attributes.
+   * @param options Options for narrowing the search
+   * @return a `HarnessPredicate` configured with the given options.
+   */
+  static with(options: TabGroupHarnessFilters = {}): HarnessPredicate<MatTabGroupHarness> {
+    return new HarnessPredicate(MatTabGroupHarness);
```
Comment (jelbourn) [inline src/material-experimental/mdc-tabs/harness/tabs-harness.ts]: For a follow up: I could see having a filter for the selected tab label

File: src/material-experimental/mdc-tabs/harness/tabs-harness.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    for (let tab of await this.getTabs()) {
+      if (await tab.isSelected()) {
+        return tab;
+      }
+    }
+    throw new Error('No selected tab could be found.');
+  }
+}
+
+export class MatTabHarness extends ComponentHarness {
```
Comment (jelbourn) [inline src/material-experimental/mdc-tabs/harness/tabs-harness.ts]: Can this go in its own file?

Also needs JsDoc description

File: src/material-experimental/mdc-tabs/harness/tabs-harness.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+  /** Gets all tabs of the tab group. */
+  async getTabs(): Promise<MatTabHarness[]> {
+    return this._tabs();
+  }
+
+  /** Gets the selected tab of the tab group. */
+  async getSelectedTab(): Promise<MatTabHarness> {
+    for (let tab of await this.getTabs()) {
+      if (await tab.isSelected()) {
```
Comment (mmalerba) [inline src/material-experimental/mdc-tabs/harness/tabs-harness.ts]: Promise.all

Comment (devversion) [reply src/material-experimental/mdc-tabs/harness/tabs-harness.ts]: where you want to use `Promise.all` here? Also isn't it faster to only check sequentially and exit earlier without awaiting remaining tabs?

File: src/material-experimental/mdc-tabs/harness/tabs-harness.ts (hunk 2 above)
Comment (devversion) [reply src/material-experimental/mdc-tabs/harness/tabs-harness.ts]: Yes, should be possible to live in a separate file. Good point with the JSDoc description.

File: src/material-experimental/mdc-tabs/harness/tabs-harness.ts (hunk 3 above)
Comment (mmalerba) [reply src/material-experimental/mdc-tabs/harness/tabs-harness.ts]: I think `Promise.all` parallelizes them, so it's better to do that than do them sequentially and find out that its the last one

Comment (devversion) [reply src/material-experimental/mdc-tabs/harness/tabs-harness.ts]: That's a question of probability, but I assume that your proposal has a higher chance being faster. will update :smile:

Comment (mmalerba) [reply src/material-experimental/mdc-tabs/harness/tabs-harness.ts]: it has a better worst-case

Comment (devversion) [reply src/material-experimental/mdc-tabs/harness/tabs-harness.ts]: true, done! please have another look

**Comment by devversion** [comment]
Ops. looks like I forgot pushing the new file for the tab-label class. Pushed now.

**Pull request merged**